global/scripts/MyelinMap_BC.sh

```diff
@@ -0,0 +1,188 @@
+#!/bin/bash
+set -eu
+
+pipedirguessed=0
+if [[ "${HCPPIPEDIR:-}" == "" ]]
+then
+    pipedirguessed=1
+    #fix this if the script is more than one level below HCPPIPEDIR
+    export HCPPIPEDIR="$(dirname -- "$0")/../.."
+fi
+
+source "$HCPPIPEDIR/global/scripts/newopts.shlib" "$@"
+source "$HCPPIPEDIR/global/scripts/debug.shlib" "$@"         # Debugging functions; also sources log.shlib
+
+defaultSigma=$(echo "sqrt(200)" | bc -l)
+
+#description to use in usage - syntax of parameters is now explained automatically
+opts_SetScriptDescription "implements CIFTI-based Myelin bias fields from native mesh to low resolution then back to native mesh"
+
+#mandatory
+#general inputs
+opts_AddMandatory '--study-folder' 'StudyFolder' 'path' "folder that contains all subjects"
+opts_AddMandatory '--subject' 'Subject' '100206' "one subject ID"
+opts_AddMandatory '--registration-name' 'RegName' 'MSMAll' "the registration string corresponding to the input files, e.g. 'MSMAll' or 'MSMSulc' or ''"
+opts_AddMandatory '--low-res-mesh' 'LowResMesh' 'meshnum or native' "the low resolution mesh node count (in thousands), like '164' for 164k_fs_LR, '32' for 32k_fs_LR"
+#optional inputs
+opts_AddOptional '--use-ind-mean' 'UseIndMean' 'YES or NO' "whether to use the mean of the subject's myelin map as reference map's myelin map mean , defaults to 'NO'" 'NO'
+opts_AddOptional '--mcsigma' 'CorrectionSigma' 'number' "myelin map bias correction sigma, default '$defaultSigma'" "$defaultSigma"
+opts_ParseArguments "$@"
+
+if ((pipedirguessed))
+then
+    log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
+
+# Verify required environment variables are set and log value
+log_Check_Env_Var HCPPIPEDIR
+log_Check_Env_Var CARET7DIR
+
+#display HCP Pipeline version
+log_Msg "Showing HCP Pipelines version"
+"${HCPPIPEDIR}"/show_version --short
+
+#display the parsed/default values
+opts_ShowValues
+
+# ------------------------------------------------------------------------------
+#  Main processing of script.
+# ------------------------------------------------------------------------------
+log_Msg "Starting main functionality"
+Caret7_Command=${CARET7DIR}/wb_command
+UseIndMeanBool=$(opts_StringToBool "$UseIndMean")
+# default folders
+SubjFolder=${StudyFolder}/${Subject}
+log_Msg "SubjFolder: $SubjFolder"
+AtlasSpaceFolder=${SubjFolder}/MNINonLinear
+log_Msg "AtlasSpaceFolder: $AtlasSpaceFolder"
+T1wFolder=${SubjFolder}/T1w
+log_Msg "T1wFolder: $T1wFolder"
+NativeFolder=${AtlasSpaceFolder}/Native
+log_Msg "NativeFolder: $NativeFolder"
+NativeT1wFolder=${T1wFolder}/Native
+log_Msg "NativeT1wFolder: $NativeT1wFolder"
+
+case "$LowResMesh" in
+	(164)
+		LowResMeshString=164k_fs_LR
+		LowResFolder=${AtlasSpaceFolder}
+		LowResT1wFolder=${AtlasSpaceFolder}
+		LowResBCMapName="MyelinMap"
```

Comment:
This seems wrong for the name of the resampled group myelin map, it looks like the name of the individual myelin map.  Did we ever generate a file with this name previously, and was it really the group average map?

Reply:
This version has some substantial logic mistakes. I have updated it in the latest commit.